### PR TITLE
Refactoring/23 introduce code format check and linter check

### LINKED ISSUE #23: Introduce code format check and linter check
- Introduce a check of the code formatting used.
  - Try use [pylint](https://pypi.org/project/pylint/)
  - Propose a better solution if found. (keep an eye on the licence)

- Checks have to run both on local and CI.

### PR DESCRIPTION
**Pull request by MobiTikula**
Introducing Pylint into project for better code quality. Implement all the code suggestion and get 10/10 score if it is possible.

Closes #23

### CODE REVIEW

**Comment by MobiTikula** [comment]
Most of the code suggestions are sucessfully implemented.
At this point I did ignore all the docstring documentaion errors (--disable=missing-module-docstring,missing-function-docstring,missing-class-docstring) since it is part of the next issue.
There is one exeption, I made in the code it is with GithubRateLimiter I left a comment, why I think we should keep it the way, we have it.

**Comment by MobiTikula** [comment]
Release notes
- Pylint suggestions implemented in the project

File: living_documentation_generator/generator.py [hunk 1]
```diff
@@ -4,17 +4,19 @@
 
 from datetime import datetime
 
-from github import Github
+from github import Github, Auth
 from github.Issue import Issue
 
+
```
Comment (miroslavpojer) [inline living_documentation_generator/generator.py:10]: Two empty lines are not required.

**Review by miroslavpojer — COMMENTED** [review]
- [x] pull
- [x] run
- [x] code review

I am missing these parts:
- missing section on readme - how to use pylint
  - add note about disable feature of pylint
-  missing GH workflow to run pylint as check in PR

File: living_documentation_generator/generator.py (hunk 1 above)
Comment (MobiTikula) [reply living_documentation_generator/generator.py]: Satisfied.

File: requirements.txt [hunk 2]
```diff
@@ -1,3 +1,4 @@
 requests~=2.31.0
 typing_extensions~=4.12.2
 PyGithub~=2.3.0
+pylint~=3.2.6
```
Comment (miroslavpojer) [inline requirements.txt:4]:
```suggestion
pylint~=3.2.6

```

**Review by miroslavpojer — COMMENTED** [review]
- [ ] pull
- [ ] code review
- [ ] run pylint

Could you please merge your yml file and inner pylint solution into origin one?

**Review by miroslavpojer — APPROVED** [review]

**Pull request merged**